autoflip_calculators/text_detection_calculator.cc

```diff
@@ -109,21 +101,46 @@ ::mediapipe::Status TextDetectionCalculator::GetContract(
     mediapipe::CalculatorContract* cc) {
   cc->Inputs().Tag(kInputVideo).Set<ImageFrame>();
   cc->Outputs().Tag(kOutputRegion).Set<DetectionSet>();
+
   return ::mediapipe::OkStatus();
 }
 
 ::mediapipe::Status TextDetectionCalculator::Open(
     mediapipe::CalculatorContext* cc) {
-  options_ = cc->Options<TextDetectionCalculatorOptions>();
-  scorer_ = absl::make_unique<VisualScorer>(options_.scorer_options());
-  detector_ = cv::dnn::readNet(kModel_path_);
+  const auto& options = cc->Options<TextDetectionCalculatorOptions>();
+  scorer_ = absl::make_unique<VisualScorer>(options.scorer_options());
+  use_visual_scorer_ = options.use_visual_scorer();
+  confidence_threshold_ = options.confidence_threshold();
+  nms_threshold_ = options.nms_threshold();
+  RET_CHECK(!options.model_path().empty())
+      << "Model path in options is required.";
+  std::string model_path = options.model_path();
+  detector_ = cv::dnn::readNet(model_path);
 
   return ::mediapipe::OkStatus();
 }
 
-void TextDetectionCalculator::DecodeBoundingBoxes(const cv::Mat& scores, const cv::Mat& geometry, float scoreThresh,
-                         std::vector<cv::RotatedRect>& detections, std::vector<float>& confidences) {
-    detections.clear();
+void TextDetectionCalculator::DetectText(const cv::Mat& frame, cv::Mat* scores, cv::Mat* geometry) {
+  cv::Mat blob;
+  // Mean subtraction and scalling.
+  // Details for blobFromImage: https://docs.opencv.org/3.4/d6/d0f/group__dnn.html#ga98113a886b1d1fe0b38a8eef39ffaaa0.
+  cv::dnn::blobFromImage(frame, blob, kSCALE_FACTOR, cv::Size(EAST_WIDTH, EAST_HEIGHT), 
+                cv::Scalar(kMEAN_B, kMEAN_G, kMEAN_R), true, false);
+  // Detect the text.
+  detector_.setInput(blob);
+  std::vector<cv::Mat> outs;
+  std::vector<cv::String> outNames(2);
+  outNames[0] = "feature_fusion/Conv_7/Sigmoid";
+  outNames[1] = "feature_fusion/concat_3";
+  detector_.forward(outs, outNames);
+
+  *scores = outs[0];
+  *geometry = outs[1];
+}
+
+void TextDetectionCalculator::DecodeBoundingBoxes(const cv::Mat& scores, const cv::Mat& geometry, const float scoreThresh,
+                        std::vector<cv::RotatedRect>* detections, std::vector<float>* confidences) {
+    detections->clear();
     CV_Assert(scores.dims == 4);
```

Comment:
Use RET_CHECK here, we can't use Assert in mediapipe because it will crash the server.  Change return type of the function to ::mediapipe::Status